core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java

```diff
@@ -113,6 +116,157 @@ public void testFailureEqualsAndHashcode() {
         checkEqualsAndHashCode(randomFailure(), copy, mutate);
     }
 
+    public void testShardInfoToXContent() throws IOException {
+        ReplicationResponse.ShardInfo shardInfo = new ReplicationResponse.ShardInfo(5, 3);
+
+        final XContent xContent = randomFrom(XContentType.values()).xContent();
+        try (XContentBuilder builder = XContentBuilder.builder(xContent)) {
+            builder.startObject();
+            shardInfo.toXContent(builder, ToXContent.EMPTY_PARAMS);
+            builder.endObject();
+
+            // Expected JSON is {"_shards":{"total":5,"successful":3,"failed":0}}
+            try (XContentParser parser = xContent.createParser(builder.bytes())) {
+                assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+                assertEquals("_shards", parser.currentName());
+                assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+                assertEquals("total", parser.currentName());
+                assertEquals(XContentParser.Token.VALUE_NUMBER, parser.nextToken());
+                assertEquals(shardInfo.getTotal(), parser.intValue());
+                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+                assertEquals("successful", parser.currentName());
+                assertEquals(XContentParser.Token.VALUE_NUMBER, parser.nextToken());
+                assertEquals(shardInfo.getSuccessful(), parser.intValue());
+                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+                assertEquals("failed", parser.currentName());
+                assertEquals(XContentParser.Token.VALUE_NUMBER, parser.nextToken());
+                assertEquals(shardInfo.getFailed(), parser.intValue());
+                assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+                assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+                assertNull(parser.nextToken());
+            }
+        }
+    }
+
+    public void testRandomShardInfoToXContent() throws IOException {
+        final ReplicationResponse.ShardInfo shardInfo = randomShardInfo();
+
+        final XContent xContent = randomFrom(XContentType.values()).xContent();
+        try (XContentBuilder builder = XContentBuilder.builder(xContent)) {
+            builder.startObject();
+            shardInfo.toXContent(builder, ToXContent.EMPTY_PARAMS);
+            builder.endObject();
```

Comment:
I am adding a util method for this in XContentHelper, maybe worth replacing this later (should not block this PR)